File: code/implementation/Subscriber.cpp
#include "Subscriber.h"
#include <string>
#include <vector>

using namespace std;

Subscriber::Subscriber(ShippingAddress address, string billingName, string billingAddress, int startingCopies)
{
    addresses.push_back(address);
    this->billingName = billingName;
    this->billingAddress = billingAddress;
    copiesRemaining = startingCopies;

}

void Subscriber::addAddress(ShippingAddress address)
{
    addresses.push_back(address);
}

vector<ShippingAddress> Subscriber::getAddresses() const
{
    return addresses;
}

vector<string> Subscriber::getShippingLables()
{
    vector<string> labels;
    for(int i = 0; i < addresses.size(); ++i)
    {
        labels.push_back(addresses.at(i).getName() + "\n" + addresses.at(i).getAddress());
    }
    return labels;
}

File: NewJavaCode/Subscriber.java
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;


public class Subscriber implements Serializable{


	private static final long serialVersionUID = 1786967556061021051L;
	protected List<ShippingAddress> addresses;
	protected int copiesRemaining;
	protected String billingName;
	protected String billingAddress;
	
	public Subscriber(String billingName, String billingAddress, int startingCopies){
		 
		this.billingName = billingName;
		this.billingAddress = billingAddress;
		if(startingCopies < 0)
			startingCopies = 0;
		this.copiesRemaining = startingCopies;
		addresses = new ArrayList<ShippingAddress>();
	 }
	
	public void addAddress(ShippingAddress address){
		addresses.add(address);
	}
	public List<ShippingAddress> getAddresses(){
		return addresses;
	}

    public int getCopiesRemaining() {
        return copiesRemaining;
    }
    
    public void increaseCopiesRemainingBy(int val){
         this.copiesRemaining += val;
    }
    
    public void incCopiesRemaining(){
         copiesRemaining++;
    }
    
    public boolean decCopiesRemaining(){
         return --copiesRemaining == 0;
    }
    
    public void setBillingAddress(String billingAddress){
         this.billingAddress = billingAddress;
    }

    public String getBillingAddress(){
         return billingAddress;
    }

    public void setBillingName(String billingName){
         this.billingName = billingName;
    }

    public String getBillingName(){
         return billingName;
    }

    //to-do
    List<String> getShippingLables(){
    	return null;
    }
    public String toString(){
    	return billingName;
    }
	
	
}
